teemaderegister-be/src/models/user.js

```diff
@@ -73,8 +78,9 @@ userSchema.pre('save', async function (next) {
 * Helper method for validating user's password on login through user.comparePassword
 * Important! Do not use arrow function, will lose ref to (this)
 */
-userSchema.methods.comparePassword = function (candidatePassword) {
-  return bcrypt.compare(candidatePassword, this.login.password)
+userSchema.methods.comparePassword = async function (candidatePassword) {
+  const value = await bcrypt.compare(candidatePassword, this.login.password)
+  return value
```

Comment:
this function can still return a promise to be used in await format, but keep it as is, its ok